### PR TITLE
Display Notifications and mark as read or unread

### PR DESCRIPTION
**Pull request by pJeyakumar**
1. Create a `Logbook Entry` (and be sure to pick yourself as the `first_pilot_in_command`.
2. Update your `User`.
3. Create a new `Aircraft`.
4. Go to the `Your Notifications` tab and verify that you have received notifications from the above actions.
5. Verify that any Notification that hasn't been read is blue and has a `Mark as Read` button.
6. Click the `Mark as Read` button for one of the notifications and refresh the page.
7. Verify that the notification has turned grey and has a `Mark as Unread` button. You'll also notice that notifications that are `Unread` (i.e. the blue ones) are always first in the list.
8. Click the `Mark as Unread` button and refresh. Verify that the notification turns blue and has a `Mark as Read` button.

### CODE REVIEW

File: app/controllers/notifications_controller.rb [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+      else
+        format.html { render :edit, status: :unprocessable_entity }
+        format.json { render json: @notification.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  # PATCH/PUT /notifications/1/toggle_read or /notifications/1/toggle_read.json
+  def toggle_read
+    @notification.read? ? @notification.mark_as_unread! : @notification.mark_as_read!
```
Comment (ddnguyen93) [inline app/controllers/notifications_controller.rb:52]: redirect to the notification page after toggle

File: app/views/notifications/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+          <h3 class="text-light text-center">You have no Notifications!</h3>
+        </div>
+      </div>
+    <% end %>
+    <% @notifications.each do |notification| %>
+      <div class="col-md-4 mb-4">
+        <div class="<%= notification.unread? ? "card bg-primary-subtle" : "card border" %>">
+          <div class="card-body">
+            <%= render notification %>
+            <p class="mt-2">
```
Comment (ddnguyen93) [inline app/views/notifications/index.html.erb:21]: remove?

File: app/controllers/notifications_controller.rb (hunk 1 above)
Comment (hvillero) [reply app/controllers/notifications_controller.rb:52]: You can use the same as you have in the update: 
```ruby
 respond_to do |format|
    format.html { redirect_to notification_url(@notification), notice: "Notification was successfully updated." }
    format.json { render :show, status: :ok, location: @notification }
 end
```

**Review by hvillero — CHANGES_REQUESTED** [review]

File: app/controllers/notifications_controller.rb [hunk 3]
```diff
@@ -0,0 +1,75 @@
+class NotificationsController < ApplicationController
+  before_action :set_notification, only: %i[ show edit update destroy toggle_read ]
+
+  # GET /notifications or /notifications.json
+  def index
+    @notifications = Notification.where(recipient_id: current_user.id).order(read_at: :desc)
```
Comment (ddnguyen93) [inline app/controllers/notifications_controller.rb:6]:
```suggestion
    @notifications = current_user.notifications.order(read_at: :desc)
```

**Review by ddnguyen93 — APPROVED** [review]

**Review by hvillero — APPROVED** [review]

**Review by srb- — APPROVED** [review]
:+1:

**Pull request merged**